src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <pb.h>
+
+typedef struct ProtoMulticastCommunicationProfile ProtoMulticastCommunicationProfile_t;
+
+/**
+ * Event Flags: used to signal tasks to unblock and run specific actions
+ */
+typedef enum
+{
+    PROTO_UPDATED   = 1 << 0,  // protobuf struct has been updated
+    RECEIVED_PROTO  = 1 << 1,  // new protobuf has been received
+    RECEIVE_TIMEOUT = 1 << 2,  // blocking takes longer than NETWORK_TIMEOUT_MS
+} ProtoMulticastCommunicationEventFlags_t;
+
+/***
+ * TASKS:
+ *
+ *                    Sender Task                             Listener Task
+ *         +-------------------------------+         +------------------------------+
+ *         |                               |         |                              |
+ *         |  wait for NETIF_CONFIGURED    |         |  wait for NETIF_CONFIGURED   |
+ *         |                               |         |                              |
+ *         |  join multicast group         |         |  join multicast group        |
+ *         |                               |         |                              |
+ *     +----> wait for PROTO_UPDATED event |     +----> wait for new proto on netif |
+ *     |   |                               |     |   |                              |
+ *     |   |  acquire lock                 |     |   |  acquire lock                |
+ *     |   |                               |     |   |                              |
+ *     |   |  serialize proto to buffer    |     |   |  deserialize buffer to proto |
+ *     |   |                               |     |   |                              |
+ *     |   |  release lock                 |     |   |  release lock                |
+ *     |   |                               |     |   |                              |
+ *     +----+ send buffer to group         |     +----+ notify RECEIVED_PROTO event |
+ *         |                               |         |                              |
+ *         +-------------------------------+         +------------------------------+
+ *
+ * @param arg The pointer to the allocated communication profile to use for the Task
+ * NOTE: These functions must run as a new Task
+ */
+void io_proto_multicast_sender_Task(void* arg);
+void io_proto_multicast_listener_Task(void* arg);
+
+/**
+ * LWIP automatically places MX_LWIP_Init() in a default task. That function
+ * call internally creates another tcpip task, and the default task just loops
```

Comment:
```suggestion
 * call internally creates another TCP IP task, and the default task just loops
```

Reply:
I think it's technically `TCP/IP`?

Reply:
lwip calls it `tcpip_init` and `tcpip_task` which is why I had it like that, changed to TCP/IP